programs/drift/src/instructions/user.rs

```diff
@@ -154,6 +154,34 @@ pub fn handle_initialize_user_stats(ctx: Context<InitializeUserStats>) -> Result
     Ok(())
 }
 
+pub fn handle_initialize_referrer_name(
+    ctx: Context<InitializeReferrerName>,
+    name: [u8; 32],
+) -> Result<()> {
+    let authority_key = ctx.accounts.authority.key();
+    let user_stats_key = ctx.accounts.user_stats.key();
+    let user_key = ctx.accounts.user.key();
+    let mut referrer_name = ctx
+        .accounts
+        .referrer_name
+        .load_init()
+        .or(Err(ErrorCode::UnableToLoadAccountLoader))?;
+
+    let user = load!(ctx.accounts.user)?;
+    validate!(
+        user.sub_account_id == 0,
+        ErrorCode::InvalidReferrer,
+        "must be subaccount 0"
+    )?;
```

Comment:
What if I deleted my first subaccount after creating some others (subaccount 0)? Then my wallet is forever unable to make a referral code?

Reply:
We could just make it so users can't delete the subaccount 0? I think there are might be some other bugs around that anyway, I just tested deleting it on one of my devnet accts and it showed me an oops page immediately afterwards even though it did work.